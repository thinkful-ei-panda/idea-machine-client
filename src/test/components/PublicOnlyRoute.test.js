import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import PublicOnlyRoute from '../../components/Utils/PublicOnlyRoute';


describe(`PublicOnlyRoute Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <PublicOnlyRoute exact
                    path='/register'
                    component={<div></div>}/>
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <PublicOnlyRoute exact
                        path='/register'
                        component={<div></div>}/>
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})