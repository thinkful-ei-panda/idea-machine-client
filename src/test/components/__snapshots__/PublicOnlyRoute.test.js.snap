// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicOnlyRoute Component Snapshot test Renders the UI as expected 1`] = `null`;
